binderhub/tests/test_utils.py

```diff
@@ -0,0 +1,25 @@
+from binderhub import utils
+
+
+def test_rendezvous_rank():
+    # check that a key doesn't move if its assigned bucket remains but the
+    # other buckets are removed
```

Comment:
I think this isn't strictly a guarantee made by the algorithm. It only promises to move as few keys as possible. However that is quite tricky to test, ideas welcome.

Reply:
I think this follows, and that this follows specifically due to proof of the "perfectly stable" algorithm classification.